### ISSUE TITLE
Fix tests under oraclejdk8

### ISSUE THREAD

**Issue by ryanfb** [sosol/sosol#62]
Seems to consistently result in:
```
# Running tests:

E.......E

Finished tests in 23.862000s, 0.3772 tests/s, 0.6286 assertions/s.

  1) Error:
test_: for community community testing should user creates and submits publication to community. (CommunityWorkflowTest):
NumbersRDF::Timeout: NumbersRDF::Timeout
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:213:in `collection_names'
    /home/travis/build/ryanfb/sosol/app/models/hgv_identifier.rb:24:in `collection_names_hash'
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:170:in `titleize'
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:429:in `title'
    /home/travis/build/ryanfb/sosol/app/models/publication.rb:209:in `log_info'
    org/jruby/RubyArray.java:1613:in `each'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/activerecord-3.2.18/lib/active_record/associations/collection_proxy.rb:91:in `method_missing'
    /home/travis/build/ryanfb/sosol/app/models/publication.rb:208:in `log_info'
    /home/travis/build/ryanfb/sosol/test/integration/community_workflow_test.rb:235:in `CommunityWorkflowTest'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/actionpack-3.2.18/lib/action_dispatch/testing/integration.rb:351:in `open_session'
    org/jruby/RubyKernel.java:1886:in `tap'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/actionpack-3.2.18/lib/action_dispatch/testing/integration.rb:350:in `open_session'
    /home/travis/build/ryanfb/sosol/test/integration/community_workflow_test.rb:230:in `CommunityWorkflowTest'
    org/jruby/RubyBasicObject.java:1562:in `instance_exec'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:413:in `test_: for community community testing should user creates and submits publication to community. '

  2) Error:
test_: for idp3 sosol testing should user creates and submits publication to sosol. (SosolWorkflowTest):
NumbersRDF::Timeout: NumbersRDF::Timeout
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:213:in `collection_names'
    /home/travis/build/ryanfb/sosol/app/models/hgv_identifier.rb:24:in `collection_names_hash'
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:170:in `titleize'
    /home/travis/build/ryanfb/sosol/app/models/identifier.rb:429:in `title'
    /home/travis/build/ryanfb/sosol/app/models/publication.rb:209:in `log_info'
    org/jruby/RubyArray.java:1613:in `each'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/activerecord-3.2.18/lib/active_record/associations/collection_proxy.rb:91:in `method_missing'
    /home/travis/build/ryanfb/sosol/app/models/publication.rb:208:in `log_info'
    /home/travis/build/ryanfb/sosol/test/integration/sosol_workflow_test.rb:188:in `SosolWorkflowTest'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/actionpack-3.2.18/lib/action_dispatch/testing/integration.rb:351:in `open_session'
    org/jruby/RubyKernel.java:1886:in `tap'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/actionpack-3.2.18/lib/action_dispatch/testing/integration.rb:350:in `open_session'
    /home/travis/build/ryanfb/sosol/test/integration/sosol_workflow_test.rb:177:in `SosolWorkflowTest'
    org/jruby/RubyBasicObject.java:1562:in `instance_exec'
    /home/travis/build/ryanfb/sosol/vendor/bundle/jruby/1.9/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:413:in `test_: for idp3 sosol testing should user creates and submits publication to sosol. '
```
See e.g.: https://travis-ci.org/ryanfb/sosol/jobs/67204596

Tests also seem to run much faster under oraclejdk8, so fixing this may eventually allow performance improvements in production.

**Comment by ryanfb** [comment]
In `JRubyXML::apply_xpath` (`lib/jruby_xml.rb`):
```
java.lang.RuntimeException: XPathFactory#newInstance() failed to create an XPathFactory for the default object model: http://java.sun.com/jaxp/xpat
h/dom with the XPathFactoryConfigurationException: javax.xml.xpath.XPathFactoryConfigurationException: java.util.ServiceConfigurationError: javax.xml.xpath.XPathFactory: jar:file:/root/sosol/lib/
java/saxon9he.jar!/META-INF/services/javax.xml.xpath.XPathFactory:2: Illegal configuration-file syntax
```
See: https://saxonica.plan.io/issues/1944

Potentially fixable by bumping our Saxon JAR?

**Comment by ryanfb** [comment]
Fixed @ 548dc7e93f1b509d002d421138c04bd5e78acec0.
